src/current/v23.2/plpgsql.md

```diff
@@ -0,0 +1,417 @@
+
+	~~~ sql
+	RAISE level USING MESSAGE = 'message';
+	~~~
+
+- `expression` is an expression to display that corresponds to the specified `option`. If `ERRCODE` is the specified option, this must be a valid [SQLSTATE error code or name](https://www.postgresql.org/docs/16/errcodes-appendix.html).
+
+For example:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE OR REPLACE PROCEDURE raise_time() AS $$
+  BEGIN
+    RAISE NOTICE 'current timestamp: %', now()
+	USING HINT = 'Call this procedure again for a different result';
+  END
+  $$ LANGUAGE PLpgSQL;
+~~~
+
+{% include_cached copy-clipboard.html %}
+~~~
+CALL raise_time();
+~~~
+
+~~~
+NOTICE: current timestamp: 2024-01-05 23:09:08.0601+00
+HINT: Call this procedure again for a different result
+CALL
+~~~
+
+#### Write exception logic
+
+Use an `EXCEPTION` statement to catch and handle specified errors. Any valid [SQLSTATE error code or name](https://www.postgresql.org/docs/16/errcodes-appendix.html) can be specified, other than `query_canceled`, `assert_failure`, or Class 40 (transaction rollback) errors. If a specified error is caught, the exception handling statements are executed. Any unspecified errors are caught by `WHEN OTHERS`, if included.
+
+~~~ sql
+EXCEPTION
+	WHEN error THEN
+		handle_exception;
+	[ WHEN error_n THEN
+		handle_exception_n; ]
+	[ WHEN OTHERS THEN
+		handle_other_exceptions; ]
+~~~
+
+`EXCEPTION` logic is typically included after the main body of a PL/pgSQL block. For example:
+
+~~~ sql
+BEGIN
+	...
+  EXCEPTION
+    WHEN not_null_violation THEN
+      RETURN 'not_null_violation';
+    WHEN OTHERS THEN
+      RETURN 'others';
+  END
+~~~
+
+## Examples
+
+### Create a user-defined function using PL/pgSQL
+
+{% include {{ page.version.version }}/sql/udf-plpgsql-example.md %}
+
+### Create a stored procedure using PL/pgSQL
+
+{% include {{page.version.version}}/sql/movr-statements.md %}
+
+{% include {{ page.version.version }}/sql/stored-proc-example.md %}
+
+## Known limitations
+
+- PL/pgSQL blocks cannot be nested.
+- `FOR` loops (including `FOR` cursor loops) are not supported.
+- Cursors used in PL/pgSQL always materialize their contents eagerly on opening. This can affect performance and resource usage.
+- PL/pgSQL `EXCEPTION` blocks cannot catch [transaction retry errors]({% link {{ page.version.version }}/transaction-retry-error-reference.md %}).
+- PL/pgSQL arguments cannot be referenced with ordinals (e.g., `$1`, `$2`).
+- `RECORD` and `ROW`-type variables cannot be declared in PL/pgSQL.
+- `NOT NULL` variables cannot be declared in PL/pgSQL.
+- When using the `RAISE` statement, schema objects related to the error cannot be named using `COLUMN`, `CONSTRAINT`, `DATATYPE`, `TABLE`, and `SCHEMA`.
+- The `INTO` statement in PL/pgSQL does not support the `STRICT` option.
+- The `PERFORM` keyword is not supported for PL/pgSQL.
```

Comment:
Some other unsupported features:
Entirely unsupported statements: https://github.com/cockroachdb/cockroach/issues/117744
EXIT and CONTINUE labels and conditions: https://github.com/cockroachdb/cockroach/issues/115271
Variable shadowing: https://github.com/cockroachdb/cockroach/issues/117508
Cursor declaration with parameters: https://github.com/cockroachdb/cockroach/issues/117746

Reply:
Added a bunch of these!